fuzz/block_fuzz.go

```diff
@@ -82,25 +82,24 @@ func beaconFuzzBlockZrnt(bb []byte, sb []byte) ([]byte, bool) {
 		panic("Deadline exceeded")
 	})
 
-	st := &phase0.BeaconState{}
-	if err := zssz.Decode(bytes.NewReader(sb), uint64(len(sb)), st, phase0.BeaconStateSSZ); err != nil {
+	st := &beacon.BeaconState{}
+	if err := zssz.Decode(bytes.NewReader(sb), uint64(len(sb)), st, beacon.BeaconStateSSZ); err != nil {
 		return fail(err)
 	}
-	blk := &phase0.SignedBeaconBlock{}
-	if err := zssz.Decode(bytes.NewReader(bb), uint64(len(bb)), blk, phase0.SignedBeaconBlockSSZ); err != nil {
+	blk := &beacon.SignedBeaconBlock{}
+	if err := zssz.Decode(bytes.NewReader(bb), uint64(len(bb)), blk, beacon.SignedBeaconBlockSSZ); err != nil {
 		return fail(err)
 	}
-	ffstate := phase0.NewFullFeaturedState(st)
-	ffstate.LoadPrecomputedData()
-	blockProc := new(phase0.BlockProcessFeature)
-	blockProc.Meta = ffstate
-	blockProc.Block = blk
-	if err := ffstate.StateTransition(blockProc, true /*validate state root*/); err != nil {
+	state, err := beacon.AsBeaconStateView(beacon.BeaconStateType.Deserialize(bytes.NewReader(bb), uint64(len(bb))))
+	if err != nil {
+		return fail(err)
+	}
+	if err := state.StateTransition(ctx, nil, blk, false /*TODO:something*/); err != nil {
```

Comment:
Let me know if you need help using ZRNT.

In this case here, you'll need an epochs context for the state transition. This type of context is re-used in consecutive block transitions, and automatically updated as part of the transition. It precomputes things like the committee shuffling and proposers. And can be efficiently forked (`.copy()` is shallow, but safe for forked use).

You can load one from the state using:
```go
epc, err := state.NewEpochsContext()
```
And then call the transition as:
```go
err = state.StateTransition(ctx, epc, blk, false)
```
And I'm not sure if it should verify the state root or not, it depends on the fuzzing approach. If that root is fuzzed, it may be better to just ignore it. And the block signature too. The beacon-fuzz team probably has some thoughts how to deal with signatures and merkle roots etc.

Reply:
Thanks for the review. I've added an epochs context and set validateResult to true.